app/models/host/discovered.rb

```diff
@@ -41,8 +41,12 @@ def self.import_host_and_facts facts
     # construct hostname
     prefix_from_settings = Setting[:discovery_prefix]
     hostname_prefix = prefix_from_settings if prefix_from_settings.present? && prefix_from_settings.match(/^[a-zA-Z].*/)
-    hostname_prefix ||= 'mac'
-    hostname = FacterUtils::bootif_mac(facts).try(:downcase).try(:gsub,/:/,'').try(:sub,/^/, hostname_prefix)
+
+    name_fact = return_first_valid_fact(Setting::Discovered.discovery_hostname_fact_array,facts)
+    raise(::Foreman::Exception.new(N_("Invalid facts: hash does not contain the required fact '%s'"), name_fact)) unless name_fact && name_fact.present?
```

Comment:
Also the i18n is incorrect, see http://projects.theforeman.org/projects/foreman/wiki/Translating